### DISCUSSION TITLE
Fetching number of ticks, domain, scales that a given plot is currently using.

### DISCUSSION THREAD

**Discussion by thedivtagguy** [observablehq/plot#2148] [Q&A] [answered]
Does Plot have any way to return the exact number of ticks (and possibly what they are, as an array) that are being displayed, or the domain and other things that it has calculated internally? Is this information exposed? 

![image](https://github.com/user-attachments/assets/88eb0938-62aa-4b4d-87de-99b66d139203)

For example in the plot above, I'd like to be able to see what the number of ticks (if I have applied an interval as well) which are on the x and y-axis. My question can also be perhaps extended to asking if I _can_ retrieve this information, what else is exposed?

**Comment by thedivtagguy** [comment, marked as answer]
Ooh just found issue #741 and learnt that I can do this: 

```js
Plot.plot(plotOptions).scale("y")
Plot.plot(plotOptions).scale("x")
```

And I _almost_ get what I need. Fantastic. But how can I still fetch the actual length of ticks and what they are? Is there a `Plot.plot(plotOptions).scale("y").ticks()`?

Is this mentioned somewhere in the documentation that I might have missed? 
What else is can be accessed?

**Reply by Fil** [reply]
https://observablehq.com/plot/features/plots#plot_scale

**Reply by thedivtagguy** [reply]
Thank you! Could you point me to this part of the question: 

> But how can I still fetch the actual length of ticks and what they are?

**Reply by mbostock** [reply]
The generated ticks are not currently exposed via an API, but you can inspect the generated SVG to see which ticks were generated.

Why do you want to know the generated tick values?

**Reply by thedivtagguy** [reply]
> Why do you want to know the generated tick values?

@mbostock I am building some internal tooling and wanted to log whatever I can as metadata about the chart. Since `interval` does some calculations, it would be helpful to be able to fetch what the final output is programmatically. 

Another use-case for wanting to know the exact length of the ticks array is having a controllable input for setting the number of visible ticks. Since these are generated charts, I would like to be able to tune the input to the exact values instead of setting a general `min` & `max`. If this is inferable from the current `scale()` output in some way, that would be equally good enough!  

> The generated ticks are not currently exposed via an API, but you can inspect the generated SVG to see which ticks were generated.

Otherwise yes, might just resort to this.
